api/client/client.go

```diff
@@ -25,38 +25,25 @@ import (
 	"github.com/docker/compose-cli/api/containers"
 	apicontext "github.com/docker/compose-cli/api/context"
 	"github.com/docker/compose-cli/api/context/store"
+	"github.com/docker/compose-cli/api/errdefs"
 	"github.com/docker/compose-cli/api/resources"
 	"github.com/docker/compose-cli/api/secrets"
 	"github.com/docker/compose-cli/api/volumes"
 )
 
 // New returns a backend client associated with current context
 func New(ctx context.Context) (*Client, error) {
-	return newWithDefaultBackend(ctx, "")
-}
-
-// NewWithDefaultLocalBackend returns a backend client associated with current context or local backend if on default context type
-func NewWithDefaultLocalBackend(ctx context.Context) (*Client, error) {
-	return newWithDefaultBackend(ctx, store.LocalContextType)
-}
-
-func newWithDefaultBackend(ctx context.Context, defaultBackend string) (*Client, error) {
-	currentContext := apicontext.CurrentContext(ctx)
-	s := store.ContextStore(ctx)
+	currentContext := apicontext.Current()
+	s := store.Instance()
 
 	cc, err := s.Get(currentContext)
 	if err != nil {
 		return nil, err
 	}
 
-	backendName := cc.Type()
-	if backendName == store.DefaultContextType && defaultBackend != "" {
-		backendName = defaultBackend
-	}
-
-	service, err := backend.Get(ctx, backendName)
-	if err != nil {
-		return nil, err
+	service := backend.Current()
```

Comment:
confirmed : would be more explicit (and more go-like) returning a potential error here, rather than letting the caller check if nil